### DISCUSSION TITLE
Support for Mapbox CSP

### DISCUSSION THREAD

**Discussion by blackswanny** [uber/react-map-gl#2397] [Q&A]
Mapbox uses Web Worker so it can experience issues with Content Security Policy if web worker is hosted in another host than the page that loads it
Mapbox provides solution to use another scripts. However, I don't see how to do the same via react-map-gl to point to specific mapbox-gl-csp.js file as source of Mapbox? react-map-gl  only allows to change worker URL or Class, which may not be enough to avoid issues

See
https://docs.mapbox.com/mapbox-gl-js/guides/browsers-and-testing/#strict-csp-environments

**Comment by Pessimistress** [comment]
[mapLib](https://visgl.github.io/react-map-gl/docs/api-reference/map#maplib)
